src/main/java/org/generationcp/middleware/api/germplasm/GermplasmService.java

```diff
@@ -128,4 +127,8 @@
 
 	void updateGermplasmPedigree(Integer gid, ProgenitorsUpdateRequestDto progenitorsUpdateRequstDto);
 
+	Germplasm getGermplasmWithPreferredName(Integer gid);
+
+	List<Germplasm> getProgenitorsWithPreferredName(final Integer gid);
```

Comment:
I don't think this should be in `GermplasmService` but rather in `GermplasmPedigreeService`.

Reply:
Done, getProgenitorsWithPreferredName is removed as it is not needed.